### ISSUE TITLE
[oracle] language java name leads to 'Found unparsable section' error

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#7938]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wanted to run the linter on a file with the following example. This leads to the described error.

### Expected Behaviour

The example should be parsed correctly.

### Observed Behaviour

I get the following message:

>L:   2 | P:   4 |  PRS | Line 2, Position 4: Found unparsable section: "language
                       | java name 'exClass.someMethod(j..."

### How to reproduce

```sql
function ex_func(some_string in varchar2) return raw
as language java name 'exClass.someMethod(java.lang.String) return byte[]';
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
